app/controllers/batches/print_controller.rb

```diff
@@ -27,6 +27,16 @@ def current_resource
     @batch ||= Batch.find(params[:id])
   end
 
+  def printer
+    @printer ||= Printer.find_by!(name: params[:printer])
+  end
+
+  def print_service
+    @print_service ||= PrintService.new(
+      print_strategy_factory: LabelPrinting::PrintStrategyFactory.new(config: Rails.configuration.print_service)
+    )
+  end
+
   def print_params
     params.permit(:label_type_id, :printer).merge(batch: current_resource)
```

Comment:
Is label_type_id on the incoming request still used? Or do you only used the label_type_id specified on the printer?

Reply:
Only the label_type_id specified on the printer is used, so this can be removed. Good spot.